webknossos/webknossos/geometry/bounding_box.py

```diff
@@ -14,7 +14,11 @@
 class BoundingBox:
     topleft: Vec3Int = attr.field(converter=Vec3Int)
     size: Vec3Int = attr.field(converter=Vec3Int)
-    bottomright = attr.field(init=False)
+    bottomright: Vec3Int = attr.field(init=False)
+    name: Optional[str] = attr.field(default="Unnamed Bounding Box")
+    is_visible: bool = attr.field(default=True)
+    id: Optional[str] = attr.field(default=None)
+    color: Optional[Tuple[float, float, float, float]] = attr.field(default=None)
```

Comment:
PS: Defaults can (and IMO should) be given using the standard syntax:
```suggestion
    name: Optional[str] = "Unnamed Bounding Box"
    is_visible: bool = True
    id: Optional[str] = None
    color: Optional[Tuple[float, float, float, float]] = None
```

Reply:
> Do you think it makes sense to have those attributes in all cases (even with their default values)? Having just a single class as you propose might make it easier for users, since there's no distinction between the two.

Yes, I think having only one BB class is way easier to use (if the parameters are not used, they don't hurt, either). The only downside I see would be performance, but the number of bounding boxes is usually **very** low in comparison to the actual EM data. So, this argument isn't really strong for me. Do you have other points against a unified class?

Reply:
Totally agree :+1: The only point I could imagine is that we use the data-structure for some algorithm, similar to Vec3Int in the flood-fill, and run into performance problems then. But let's solve this as it appears, I think we can still add this abstraction for exactly that purpose then (it wouldn't even need to be user-facing).